common/src/main/java/net/whimxiqal/journey/data/DataManagerImpl.java

```diff
@@ -41,80 +31,74 @@
 import net.whimxiqal.journey.data.sql.SqlPublicWaypointManager;
 import net.whimxiqal.journey.data.sql.mysql.MySqlConnectionController;
 import net.whimxiqal.journey.data.sql.sqlite.SqliteConnectionController;
-import net.whimxiqal.journey.util.Initializable;
+import net.whimxiqal.journey.data.version.MysqlDataVersionHandler;
+import net.whimxiqal.journey.data.version.SqliteDataVersionHandler;
 
-/**
- * Implementation of the {@link DataManager} in Spigot Minecraft.
- */
 public class DataManagerImpl implements DataManager {
 
-  public static final String DATABASE_FILE_NAME = "journey.db";
   private PersonalWaypointManager personalWaypointManager;
   private PublicWaypointManager publicWaypointManager;
   private PathRecordManager pathRecordManager;
   private TunnelDataManager tunnelDataManager;
 
+  public static final String DATABASE_FILE_NAME = "journey.db";
+
+  public static final String VERSION_FILE_NAME = "journeydb.ver";
+
+  public static final String VERSION_TABLE_NAME = "journey_db_version";
+
+  public static final String VERSION_COLUMN_NAME = "db_version";
+
+  private DataVersion databaseVersion = null;
+
   @Override
   public void initialize() {
-    boolean setupSchema = version() == DataVersion.V000;
+    DataVersion currentVersion;
+    int currentVersionNumber;
+
     switch (Settings.STORAGE_TYPE.getValue()) {
       case SQLITE:
         String sqliteAddress = "jdbc:sqlite:" + Journey.get().proxy().dataFolder() + "/" + DATABASE_FILE_NAME;
         SqliteConnectionController sqliteController = new SqliteConnectionController(sqliteAddress);
+
+        SqliteDataVersionHandler sqliteDataVersionHandler = new SqliteDataVersionHandler(sqliteController);
+
+        currentVersion = sqliteDataVersionHandler.version();
+        currentVersionNumber = currentVersion.internalVersion;
+
         personalWaypointManager = new SqlPersonalWaypointManager(sqliteController);
         publicWaypointManager = new SqlPublicWaypointManager(sqliteController);
         pathRecordManager = new SqlPathRecordManager(sqliteController);
         tunnelDataManager = new SqlTunnelDataManager(sqliteController);
-        if (setupSchema) {
-          try (Connection connection = sqliteController.establishConnection()) {
-            Statement statement = connection.createStatement();
-            addBatchesToStatement("/data/sql/schema/sqlite.sql", statement);
-            statement.executeBatch();
-          } catch (SQLException e) {
-            setupSchema = false;
-            e.printStackTrace();
-          }
-        }
+
+        if (currentVersionNumber < DataVersion.latest().internalVersion) {
+          databaseVersion = sqliteDataVersionHandler.runMigrations(currentVersion);
+        } else databaseVersion = DataVersion.latest();
```

Comment:
There's no real need to have `version` and `runMigrations` on the `VersionHandler` interface, since we just check what the version is and run migrations depending on that. Rather, it would be cleaner for the interface to have `updateSchema()` or something short, and even called outside of the switch statement, leveraging the interfacing nature of the version handler. Then, inside `updateSchema()`, you'd have an internal check for the version and run migrations as needed, then return the resulting version.